diff --git a/panopoly_wysiwyg.module b/panopoly_wysiwyg.module
index 0b767af..f819bed 100644
--- a/panopoly_wysiwyg.module
+++ b/panopoly_wysiwyg.module
@@ -176,10 +176,12 @@ function panopoly_wysiwyg_wysiwyg_editor_settings_alter(&$settings, $context) {
           // that we're merging, and not overwriting.
           $settings_array = array();
           foreach($settings['extended_valid_elements'] as $tag) {
-            list($tag, $allowed_attributes) = explode('[', $tag);
-            $allowed_attributes = explode('|', trim($allowed_attributes, ']'));
-            foreach($allowed_attributes as $key => $attribute) {
-              $settings_array[$tag][$attribute] = $attribute;
+            if (strpos("[", $tag) !== FALSE) {
+              list($tag, $allowed_attributes) = explode('[', $tag);
+              $allowed_attributes = explode('|', trim($allowed_attributes, ']'));
+              foreach($allowed_attributes as $key => $attribute) {
+                $settings_array[$tag][$attribute] = $attribute;
+              }
             }
           }
           $valid_elements = array();
